CvGameCoreDLL_Expansion2/CvPlot.cpp

```diff
@@ -5850,6 +5850,26 @@ void CvPlot::changeUpgradeProgress(int iChange)
 }
 
 //	--------------------------------------------------------------------------------
+fraction CvPlot::ComputeFractionalYieldFromAdjacentImprovement(CvImprovementEntry& kImprovement, YieldTypes eYield) const
+{
+	CvPlot* pAdjacentPlot = NULL;
+	fraction fRtnValue = 0;
+
+	if (kImprovement.IsYieldPerXAdjacentImprovement(eYield))
+	{
+		for(int iI = 0; iI < NUM_DIRECTION_TYPES; iI++)
+		{
+			pAdjacentPlot = plotDirection(getX(), getY(), static_cast<DirectionTypes>(iI));
+			if(pAdjacentPlot)
+			{
+				fRtnValue += kImprovement.GetYieldPerXAdjacentImprovement(eYield, pAdjacentPlot->getImprovementType());
```

Comment:
There's no IsImprovementPillaged check here and no verification that pAdjacentPlot->getImprovementType() doesn't return NO_IMPROVEMENT.

Reply:
fixed

Reply:
Fixed